bootstrap.sh

```diff
@@ -1,117 +1,308 @@
+	fi
+}
 
+do_email_setup() {
+	echo
+	email_smtp_host=
+	until [[ ${email_smtp_host} =~ ^[a-z0-9\.]*$ ]]; do
+		[[ -n "${email_smtp_host}" ]] && echo "Invalid SMTP server"
+		read -r -p "SMTP server: " email_smtp_host
+	done
+	echo
+	read -r -p "SMTP port [465]: " email_smtp_port
+	email_smtp_port=${email_smtp_port:-465}
+	echo
+	read -r -p "SMTP login: " email_login
+	echo
+	local email_password=
+	until [[ -n ${email_password} ]]; do
+		[[ -z ${email_password} ]] && echo "The password is empty"
+		read -r -s -p "SMTP password: " email_password
+	done
+	echo
+	echo
+	read -r -p "'From' e-mail: " email
+	[[ -n ${email} ]] && {
+		echo "email: \"${email}\"" |tee -a "${CUSTOM_FILE}"
+	}
+
+	read -r -p "'To' e-mail: " email_recipient
+	[[ -n "${email_recipient}" ]] && {
+		echo "email_recipient: \"${email_recipient}\"" |tee -a "${CUSTOM_FILE}"
+	}
+
+	(
+	echo "email_smtp_host: \"${email_smtp_host}\""
+	echo "email_smtp_port: \"${email_smtp_port}\""
+	echo "email_login: \"${email_login}\""
+	) |tee -a "${CUSTOM_FILE}"
+
+	if [ -z ${email_password+x} ]; then
+		echo
+	else 
+		# SECRET_FILE setup previously, save the secret
+		echo "email_password: \"${email_password}\"" >> "${SECRET_FILE}"
+	fi
 
-check_root() {
-# Check if the user is root or not
-if [[ $EUID -ne 0 ]]; then
-  if [[ ! -z "$1" ]]; then
-    SUDO='sudo -E -H'
-  else
-    SUDO='sudo -E'
-  fi
-else
-  SUDO=''
-fi
 }
 
-check_root
-# Disable interactive apt functionality
-export DEBIAN_FRONTEND=noninteractive
+get_ip_list() {
+	dig -t a +short @"${DNS_HOST}" "${1}" | grep '^[1-9]'  | tr '\n' ' '
+}
 
-# Update apt database, update all packages and install Ansible + dependencies
-$SUDO apt update -y;
-yes | $SUDO apt-get -o Dpkg::Options::="--force-confold" -fuy dist-upgrade;
-yes | $SUDO apt-get -o Dpkg::Options::="--force-confold" -fuy install software-properties-common certbot dnsutils curl git python3 python3-setuptools python3-apt python3-pip python3-passlib python3-wheel python3-bcrypt aptitude -y;
-yes | $SUDO apt-get -o Dpkg::Options::="--force-confold" -fuy autoremove;
-[ $(uname -m) == "aarch64" ] && $SUDO yes | apt install gcc dnsutils python3-dev libffi-dev libssl-dev make -y;
+install_packages_for_ansible_and_dependencies() {
+	# Disable interactive apt functionality
+	export DEBIAN_FRONTEND=noninteractive
+
+	# Update apt database, update all packages and install Ansible + dependencies
+	declare -a REQUIRED_PACKAGES=(
+	software-properties-common
+	certbot
+	dnsutils
+	curl
+	git
+	python3
+	python3-setuptools
+	python3-apt
+	python3-pip
+	python3-passlib
+	python3-wheel
+	python3-bcrypt
+	aptitude
+	)
+
+	check_root
+	"${SUDO}" apt update -y
+
+	yes | "${SUDO}" apt-get -o Dpkg::Options::="--force-confold" -fuy dist-upgrade
+
+	yes | "${SUDO}" apt-get -o Dpkg::Options::="--force-confold" -fuy install \
+		"${REQUIRED_PACKAGES[@]}"
+
+	yes | "${SUDO}" apt-get -o Dpkg::Options::="--force-confold" -fuy autoremove
+
+	aptitude show "${REQUIRED_ANSIBLE_PACKAGES[@]}"
```

Comment:
What is the purpose of this line?
aptitude show "${REQUIRED_ANSIBLE_PACKAGES[@]}"

Reply:
Informational, it isn't important.

For the aarch64 packages, why apt instead of apt-get like the other commands?